src/modules/cpu/kernel/warp_perspective.hpp

```diff
@@ -0,0 +1,2286 @@
+                locY += (perspectiveMatrix_f9->data[3] * vectorLoopCount);
+                locX += (perspectiveMatrix_f9->data[0] * vectorLoopCount);
+                srcX = (locX/locW) + roiHalfWidth;
+                srcY = (locY/locW) + roiHalfHeight;
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    compute_generic_nn_interpolation_pln_to_pln(srcY, srcX, &roiLTRB, dstPtrTemp++, srcPtrChannel, srcDescPtr, dstDescPtr);
+                    compute_warp_perspective_src_loc_next_term(vectorLoopCount, locW, locY, locX, srcY, srcX, perspectiveMatrix_f9, roiHalfHeight, roiHalfWidth);
+                }
+                dstPtrRow += dstDescPtr->strides.hStride;
+            }
+        }
+    }
+
+    return RPP_SUCCESS;
+}
+
+
+RppStatus warp_perspective_nn_f32_f32_host_tensor(Rpp32f *srcPtr,
+                                           RpptDescPtr srcDescPtr,
+                                           Rpp32f *dstPtr,
+                                           RpptDescPtr dstDescPtr,
+                                           Rpp32f *perspectiveTensor,
+                                           RpptROIPtr roiTensorPtrSrc,
+                                           RpptRoiType roiType,
+                                           RppLayoutParams srcLayoutParams,
+                                           rpp::Handle& handle)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi, roiLTRB;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+        compute_ltrb_from_xywh_host(&roi, &roiLTRB);
+        Rpp32s roiHalfWidth = roi.xywhROI.roiWidth >> 1;
+        Rpp32s roiHalfHeight = roi.xywhROI.roiHeight >> 1;
+
+        Rpp32f9 *perspectiveMatrix_f9;
+        perspectiveMatrix_f9 = (Rpp32f9 *)perspectiveTensor + batchCount;
+
+        Rpp32f *srcPtrChannel, *dstPtrChannel, *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+        srcPtrChannel = srcPtrImage;
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32s vectorIncrementPerChannel = 8;
+        Rpp32s vectorIncrementPkd = 24;
+        Rpp32u bufferLength = roi.xywhROI.roiWidth;
+        Rpp32u alignedLength = bufferLength & ~7;   // Align dst width to process 16 dst pixels per iteration
+        Rpp32s srcLoc[8] = {0};         // Since 4 dst pixels are processed per iteration
+        Rpp32s invalidLoad[8] = {0};    // Since 4 dst pixels are processed per iteration
+
+        __m256 pSrcStrideH = _mm256_set1_ps(srcDescPtr->strides.hStride);
```

Comment:
Most of the comment added for U8, should be applicable for F32, F16, I8 variants also
Please check and change accordingly

Reply:
Updated